samcli/commands/build/build_context.py

```diff
@@ -318,32 +304,47 @@ def _handle_build_pre_processing(self) -> List[Stack]:
         :return: List of modified stacks
         """
         stacks = []
-        for stack in self.stacks:
-            stacks.append(EsbuildBundlerManager(stack=stack).set_sourcemap_metadata_from_env())
-        return stacks
+        if any(EsbuildBundlerManager(stack).esbuild_configured() for stack in self.stacks):
+            # esbuild is configured in one of the stacks, will check and update stack metadata accordingly
+            for stack in self.stacks:
+                stacks.append(EsbuildBundlerManager(stack).set_sourcemap_metadata_from_env())
+            self.function_provider.update(stacks, self._use_raw_codeuri, locate_layer_nested=self._locate_layer_nested)
```

Comment:
Will `any` return `True` with first instance, or it will loop through the entire array? I was asking because we are looping over `stacks` couple of times here and was wondering if we can improve it by looping only once.

Reply:
`any` will exit early on the first `True` that it sees so it won't re-run the logic more than once.